src/create_agent/init_agent.jl

```diff
@@ -152,30 +169,26 @@ function init_agent(
         end
     end
 
-    #Go through each specified shared parameter
-    for (shared_parameter_key, dict_value) in shared_parameters
-        #Unpack the shared parameter value and the derived parameters
-        (shared_parameter_value, derived_parameters) = dict_value
-        #check if the name of the shared parameter is part of its own derived parameters
-        if shared_parameter_key in derived_parameters
-            throw(
-                ArgumentError(
-                    "The shared parameter $shared_parameter_key is also in the list of parameters to share the parameter value",
-                ),
-            )
-        end
-        agent.shared_parameters[shared_parameter_key] = SharedParameter(
-            value = shared_parameter_value,
-            derived_parameters = derived_parameters,
-        )
-    end
-
     #For each specified state
     for (state_key, state_value) in agent.states
         #Add it to the history
         agent.history[state_key] = [state_value]
     end
 
+    #Check agent for settings of shared parameters
+    check_agent(agent)
```

Comment:
check_agent kan sagtens være efter parameters er st - faktisk skal den være det for at kunne tjekke om de er blevet sat korrekt.
Så kan du ha for loopet nedenunder i for loopet ovenover som før :)